In [1]:
# Installation automatique des dépendances requises dans le noyau Jupyter actuel
%pip install -r ../requirements.txt

Note: you may need to restart the kernel to use updated packages.


# 📥 Étape 1 : Acquisition des Données & Multi-Sources (Squelette Étudiant)

Cette étape correspond au premier chapitre du pipeline de Data Science. L'objectif est d'identifier, d'importer et de consolider vos jeux de données bruts issus de différentes sources (fichiers CSV locaux, requêtes API, bases de données, etc.).

### 1. Initialisation de l'environnement

In [5]:
import os
import sys
import pandas as pd
import numpy as np

# Ajout du dossier parent au chemin de recherche des modules
sys.path.append(os.path.abspath('..'))

print("Libraries importées avec succès ! Prêt pour l'étape d'Acquisition des Données.")

Libraries importées avec succès ! Prêt pour l'étape d'Acquisition des Données.


### 2. Chargement de la source de données principale

Le jeu de données a été chargé à partir de ce dataset :

[Uncleaned Laptop Price Dataset](https://www.kaggle.com/datasets/ehtishamsadiq/uncleaned-laptop-price-dataset/data)

L'ensemble de données de ce dataset représente une collection d'annonces d'ordinateurs portables extraites d'un site de commerce électronique.

### 🔹 Date du dataset : **2019** 

### 🔹 Dimension du dataset : **1 300 ordinateurs portables** 

### 🔹 Variables catégorielles
* **marque** 
* **modèle** 
* **résolution d'écran**
* **type de processeur**
* **système d'expoitation**
* **...**

### 🔹 Variables numériques
* **taille de l'écran** 
* **mémoire** 
* **capacité de stockage**
* **prix**
* **...**


In [6]:
import kagglehub
from kagglehub import KaggleDatasetAdapter

# Load a DataFrame with a specific version of a CSV
df = kagglehub.dataset_load(
    KaggleDatasetAdapter.PANDAS,
    "unsdsn/world-happiness/versions/1",
    "2016.csv",
)

In [7]:
# Spécifiez le chemin vers votre jeu de données principal
main_data_path = '../data/raw/laptopData.csv'

# TODO: Charger le dataset à l'aide de Pandas
df_main = pd.read_csv(main_data_path)

# Affichez la taille et les premières lignes
print(f"Vérification de la taille du dataset et visualisation des premières colonnes : {df_main.shape}")
df_main.head(20)

Vérification de la taille du dataset et visualisation des premières colonnes : (1303, 12)


,Unnamed: 0,Company,TypeName,Inches,ScreenResolution,Cpu,Ram,Memory,Gpu,OpSys,Weight,Price
0,0.0,Apple,Ultrabook,13.3,IPS Panel Retina Display 2560x1600,Intel Core i5 2.3GHz,8GB,128GB SSD,Intel Iris Plus Graphics 640,macOS,1.37kg,71378.6832
1,1.0,Apple,Ultrabook,13.3,1440x900,Intel Core i5 1.8GHz,8GB,128GB Flash Storage,Intel HD Graphics 6000,macOS,1.34kg,47895.5232
2,2.0,HP,Notebook,15.6,Full HD 1920x1080,Intel Core i5 7200U 2.5GHz,8GB,256GB SSD,Intel HD Graphics 620,No OS,1.86kg,30636.0000
3,3.0,Apple,Ultrabook,15.4,IPS Panel Retina Display 2880x1800,Intel Core i7 2.7GHz,16GB,512GB SSD,AMD Radeon Pro 455,macOS,1.83kg,135195.3360
4,4.0,Apple,Ultrabook,13.3,IPS Panel Retina Display 2560x1600,Intel Core i5 3.1GHz,8GB,256GB SSD,Intel Iris Plus Graphics 650,macOS,1.37kg,96095.8080
5,5.0,Acer,Notebook,15.6,1366x768,AMD A9-Series 9420 3GHz,4GB,500GB HDD,AMD Radeon R5,Windows 10,2.1kg,21312.0000
6,6.0,Apple,Ultrabook,15.4,IPS Panel Retina Display 2880x1800,Intel Core i7 2.2GHz,16GB,256GB Flash Storage,Intel Iris Pro Graphics,Mac OS X,2.04kg,114017.6016
7,7.0,Apple,Ultrabook,13.3,1440x900,Intel Core i5 1.8GHz,8GB,256GB Flash Storage,Intel HD Graphics 6000,macOS,1.34kg,61735.5360
8,8.0,Asus,Ultrabook,14,Full HD 1920x1080,Intel Core i7 8550U 1.8GHz,16GB,512GB SSD,Nvidia GeForce MX150,Windows 10,1.3kg,79653.6000
9,9.0,Acer,Ultrabook,14,IPS Panel Full HD 1920x1080,Intel Core i5 8250U 1.6GHz,8GB,256GB SSD,Intel UHD Graphics 620,Windows 10,1.6kg,41025.6000


### Conclusion

Une première analyse du dataset est effectuée afin d’avoir une vision globale de sa structure et de son contenu.

### 🔹 Observations

* **Nombre d'enregistrements :** 1303 lignes.
* **Nombre de colonnes :** 12 colonnes.
* **Types de variables observées :**
  * des variables **catégorielles** comme la marque, le type d’ordinateur, le processeur, la carte graphique ou le système d’exploitation ;
  * des variables **numériques** comme la taille de l’écran, la mémoire RAM, le poids ou le prix.
* **Premiers constats :**
  * certaines colonnes contiennent **plusieurs informations dans une seule valeur**;
  * la colonne `Price` représente la **variable cible du projet**, puisqu’elle servira à analyser ou prédire le prix des ordinateurs.

Cette première observation permet donc **d’identifier les principales variables disponibles** et de **repérer les colonnes qui nécessiteront un nettoyage ou une transformation** avant l’analyse exploratoire et la modélisation.